### PR TITLE
Vulkan support

### PR DESCRIPTION
**Pull request by bonorumetmalorum**
Hi!

I was working on a little experiment to get back into rust. I thought it would be nice to use this library with Vulkan but realised it didn't support it. I went ahead and added the support for it. I re-created the sample app using obj-rs with Vulkan. All tests are still passing; I haven't added any new ones. Any feedback and comments would be greatly appreciated! No pressure to merge if it is unnecessary.

Thanks.

### CODE REVIEW

**Comment by simnalamburt** [comment]
Looks like your patch implements a few Traits and introduces no breaking changes. No reason to reject this PR!

File: sampleapp/Cargo.toml [hunk 1]
```diff
@@ -8,4 +8,4 @@ edition = "2018"
 
 [dependencies]
 glium = "0.26"
-obj-rs = { path = "../obj-rs", default-features = false, features = ["glium"] }
+obj-rs = { path = "../obj-rs", default-features = false, vulkano=false, features = ["glium"] }
```
Comment (simnalamburt) [inline sampleapp/Cargo.toml:11]: I get this warning from this change. Was there any reason to add this key?

```
warning: /<path>/obj-rs/sampleapp/Cargo.toml: unused manifest key: dependencies.obj-rs.vulkano
```

Comment (bonorumetmalorum) [reply sampleapp/Cargo.toml]: Nope! that's just something I forgot to remove when debugging! I'll remove the key and update the PR.

**Review by simnalamburt — APPROVED** [review]

**Comment by simnalamburt** [comment]
[Version 0.6.2 has been published.](https://crates.io/crates/obj-rs/0.6.2)

**Pull request merged**